internal/ecsservicediscovery/decoratedtask.go

```diff
@@ -72,12 +72,13 @@ func addExporterLabels(labels map[string]string, labelKey string, labelValue *st
 // Get the private ip of the decorated task.
 // Return "" when fail to get the private ip
 func (t *DecoratedTask) getPrivateIp() string {
-	if t.TaskDefinition.NetworkMode == nil || *t.TaskDefinition.NetworkMode == ecs.NetworkModeNone {
```

Comment:
When `t.TaskDefinition.NetworkMode == nil`, we _want_ the logic to fall through to the bottom of the function, where

```
if t.EC2Info != nil {
    return t.EC2Info.PrivateIP
}
```

should return the correct IP for bridge mode.